tests/test_volume.py

```diff
@@ -726,6 +734,77 @@ def test_symmetry_group_set_get(sym_group, sym_string):
         _ = Volume(data, symmetry_group=123, dtype=dtype)
 
 
+def test_symmetry_group_pass_through(symmetric_vols):
+    vol_c3, _ = symmetric_vols
+    sym_group = str(vol_c3.symmetry_group)
+    assert sym_group == "C3"
+
+    # Check symmetry_group pass-through for various transformations.
+    assert str(vol_c3.astype(np.float64).symmetry_group) == sym_group  # astype
+    assert str(vol_c3[0].symmetry_group) == sym_group  # getitem
+    assert (
+        str(vol_c3.stack_reshape((1, 1)).symmetry_group) == sym_group
+    )  # stack_reshape
+    assert (
+        str(vol_c3.downsample(vol_c3.resolution // 2).symmetry_group) == sym_group
+    )  # downsample
+
+
+def test_transformation_symmetry_warnings(symmetric_vols):
+    """
+    A warning should be emitted (once) for transpose, flip, rotate, add, sub, mult, div.
+    """
+    vol_c3, _ = symmetric_vols
+    sym_group = str(vol_c3.symmetry_group)
+    assert sym_group == "C3"
+
+    # Check we get warning on first transformation.
+    with pytest.warns(
+        UserWarning, match=r".*`symmetry_group` attribute is being set to `C1`.*"
+    ) as record:
+        vol_t = vol_c3.T
+        vol_f = vol_c3.flip()
+        vol_r = vol_c3.rotate(Rotation.about_axis("x", np.pi, dtype=vol_c3.dtype))
+    assert len(record) == 3
+
+    # Check symmetry_group has been set to C1.
+    assert str(vol_t.symmetry_group) == "C1"
+    assert str(vol_f.symmetry_group) == "C1"
+    assert str(vol_r.symmetry_group) == "C1"
+
+    # Check original volume has retained C3 symmetry.
+    assert str(vol_c3.symmetry_group) == "C3"
+
+
+def test_aglebraic_ops_symmetry_warnings(symmetric_vols):
+    vol_c3, vol_c4 = symmetric_vols
+
+    # Compatible symmetry should retain symmetry_group.
```

Comment:
Might want to programtically check that warnings are not emitted here.

Reply:
Added a check for this.